src/app/loadout-drawer/subclass-drawer/SubclassPlugDrawer.tsx

```diff
@@ -151,43 +153,76 @@ export default function SubclassPlugDrawer({
   );
 }
 
-export function getPlugsForSubclass(
+/**
+ * This creates the the plugsWithMaxSelectableSets for the plug picker and also creates sets
+ * of aspect and fragment plugs.
+ */
+function getPlugsForSubclass(
   defs: D2ManifestDefinitions | undefined,
   profileResponse: DestinyProfileResponse | undefined,
   subclass: DimItem
 ) {
-  const plugs = new Set<PluggableInventoryItemDefinition>();
+  const plugsWithMaxSelectableSets: PlugsWithMaxSelectable[] = [];
   const aspects: Set<PluggableInventoryItemDefinition> = new Set();
   const fragments: Set<PluggableInventoryItemDefinition> = new Set();
 
   if (!subclass.sockets || !defs) {
-    return { plugs, aspects, fragments };
+    return { plugsWithMaxSelectableSets, aspects, fragments };
   }
 
   for (const category of subclass.sockets.categories) {
     const sockets = getSocketsByCategoryHash(subclass.sockets, category.category.hash);
-
-    for (const socket of sockets) {
-      for (const hash of getPlugHashesForSocket(socket, profileResponse)) {
-        const plugDef = defs.InventoryItem.get(hash);
-        const isAspect = category.category.hash === SocketCategoryHashes.Aspects;
-        const isFragment = category.category.hash === SocketCategoryHashes.Fragments;
-        const isEmptySocket =
-          (isAspect || isFragment) && hash === socket.socketDefinition.singleInitialItemHash;
-
-        if (!isEmptySocket && isPluggableItem(plugDef)) {
-          plugs.add(plugDef);
-
-          if (isAspect) {
-            aspects.add(plugDef);
-          } else if (isFragment) {
-            fragments.add(plugDef);
+    // Group sockets by their plugSetHash so that we can figure out how many aspect or ability
+    // choices the user will get
+    const socketsGroupedBySetHash = _.groupBy(
+      sockets,
+      (socket) => socket.socketDefinition.reusablePlugSetHash
+    );
+
+    for (const socketGroup of Object.values(socketsGroupedBySetHash)) {
+      if (socketGroup.length) {
+        const firstSocket = socketGroup[0];
+        const plugSetHash = firstSocket.socketDefinition.reusablePlugSetHash;
+
+        if (plugSetHash && profileResponse) {
+          const plugsWithMaxSelectable: PlugsWithMaxSelectable = {
+            plugs: [],
+            plugSetHash,
+            maxSelectable: socketGroup.length,
+          };
+          // Get all the availabe plugs for the given profile
+          const plugHashes = itemsForPlugSet(profileResponse, plugSetHash).map(
```

Comment:
At some point it'd be better to use the character-specific version

Reply:
I have a TODO in there, I am trying to make this PR as small as possible as there is a lot of random stuff going on to make this change.